### PR TITLE
chore(suite): add @typescript-eslint/strict-boolean-expressions rule as warning

### PR DESCRIPTION
**Pull request by peter-sanderson**
image

`yarn nx:lint:js`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ module.exports = {
         ecmaFeatures: {
             jsx: true,
         },
+        project: ["./tsconfig.json"],
```
Comment (peter-sanderson) [inline .eslintrc.js]: This is required to have for `@typescript-eslint/strict-boolean-expressions` rule to work

File: packages/analytics/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    parserOptions: {
+        tsconfigRootDir: __dirname,
```
Comment (peter-sanderson) [inline packages/analytics/.eslintrc.js:3]: Every package needs to have this ti make sure the eslint uses the local `tsconfig` and not the global one as most of packages has some custom configurations.

File: packages/blockchain-link-types/tsconfig.json [hunk 3]
```diff
@@ -1,8 +1,17 @@
 {
     "extends": "../../tsconfig.json",
-    "compilerOptions": { "outDir": "libDev" },
+    "compilerOptions": {
+        "outDir": "libDev"
+    },
+    "include": [
+        "src"
```
Comment (peter-sanderson) [inline packages/blockchain-link-types/tsconfig.json:7]: Is this ok? I was surprised this was not here.

File: packages/e2e-utils/src/mocks/backendServer.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line @typescript-eslint/no-shadow
```
Comment (peter-sanderson) [inline packages/e2e-utils/src/mocks/backendServer.ts]: I think those errors poped up when ESLint got the type-information from typescript. But not sure.

File: packages/protobuf/tsconfig.json [hunk 5]
```diff
@@ -1,6 +1,15 @@
...
-    "compilerOptions": { "outDir": "libDev" },
-    "include": ["src"],
-    "references": [{ "path": "../schema-utils" }]
-}
+    "compilerOptions": {
+        "outDir": "libDev"
+    },
+    "include": [
+        "src",
+        "tests",
```
Comment (peter-sanderson) [inline packages/protobuf/tsconfig.json:8]: is this ok?

File: packages/e2e-utils/src/mocks/backendServer.ts (hunk 4 above)
Comment (peter-sanderson) [reply packages/e2e-utils/src/mocks/backendServer.ts]: Seems to be somehow globally imported from `typescript/lib/lib.dom.d.ts`

File: suite-native/accounts/src/components/AccountListItem.tsx [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import { useSelector } from 'react-redux';
 
 import { AccountsRootState, selectFormattedAccountType } from '@suite-common/wallet-core';
+// eslint-disable-next-line @typescript-eslint/no-shadow
```
Comment (peter-sanderson) [inline suite-native/accounts/src/components/AccountListItem.tsx:4]: Seems to be somehow globally imported from `@types/react/global.d.ts`

**Comment by peter-sanderson** [comment]
- [ ] Todo: revert the shadow-ignore commint and figure out how to not import global types

**Review by jvaclavik — APPROVED** [review]
let's be brave 

🚢 🇮🇹

**Pull request closed without merging**